### ISSUE TITLE
AWS EKS SubnetIds and SecurityIds must coexist or be both empty list.

### ISSUE THREAD

**Issue by alexkamcrypto** [aws-quickstart/quickstart-amazon-eks#210]
I am trying to use the official deployment guide to create EKS with existing VPC and subnet groups, and i am end up getting the same issue:
```
The following resource(s) failed to delete: [ALBClusterRole]. 

AWS EKS SubnetIds and SecurityIds must coexist or be both empty list.
```

Below are my configuration. Is there any problem in my parameters, or the deployment stack?

ALBIngressController | Enabled
--- | ---
AdditionalEKSAdminRoleArn | -
AdditionalEKSAdminUserArn | -
ClusterAutoScaler | Enabled
ConfigSetName | -
ConsulIntegration | Disabled
ConsulUIACMSSLCertificateArn | -
ConsulUIDomainName | -
ConsulUIHostedZoneID | -
EKSClusterName | demo-eks
EKSPublicAccessEndpoint | Disabled
EfsStorageClass | Enabled
FargateLabels | -
FargateNamespaces | -
HttpProxy | -
KeyPairName | demo_key
MaxNumberOfNodes | 3
MonitoringStack | Prometheus + Grafana
NewRelicIntegration | Disabled
NewRelicLicenseKey | -
NodeGroupOS | Amazon Linux 2
NodeGroupType | Managed
NodeInstanceFamily | Standard
NodeInstanceType | t3.medium
NumberOfNodes | 3
PerAccountSharedResources | Yes
PerRegionSharedResources | Yes
PrivateSubnet1ID | ${subnet-1a-private}
PrivateSubnet2ID | ${subnet-1b-private}
PrivateSubnet3ID | ${subnet-1c-private}
ProvisionBastionHost | Enabled
PublicSubnet1ID | ${subnet-1a-public}
PublicSubnet2ID | ${subnet-1b-public}
PublicSubnet3ID | ${subnet-1c-public}
QSS3BucketName | aws-quickstart
QSS3BucketRegion | us-east-1
QSS3KeyPrefix | quickstart-amazon-eks/
RemoteAccessCIDR | 10.200.0.0/16
SnykIntegration | Disabled
SnykIntegrationId | -
VPCID | vpc-04e459cd1e90edd1a
VaultIntegration | Disabled
VaultUIACMSSLCertificateArn | -
VaultUIDomainName | -
VaultUIHostedZoneID | -

**Comment by alexkamcrypto** [comment]
I disabled the ALB plugin setting during start up and the EKS is created smoothly. But still i cannot figure out this issue's solution.

**Comment by jaymccon** [comment]
Are your subnets tagged with the [elb tags](https://docs.aws.amazon.com/eks/latest/userguide/network_reqs.html#vpc-subnet-tagging) ?

**Comment by alexkamcrypto** [comment]
@jaymccon nope. I did not noticed this requirement from the quick start up guide.
Is it able for quick startup editing the tagging for existing subnets?

Anyway guess I can close this issue soon unless further issue in the setup.

**Comment by kn0xyland** [comment]
I noticed this issue as well, I was a little confused as the EKS docs say to use subnet tags with:
https://docs.aws.amazon.com/eks/latest/userguide/network_reqs.html#vpc-subnet-tagging

kubernetes.io/role/internal-elb = 1
kubernetes.io/role/elb = 1

Yet the EKS quick start guide says to set to:
https://aws-quickstart.github.io/quickstart-amazon-eks/#_planning_the_deployment

kubernetes.io/role/internal-elb = true
kubernetes.io/role/elb = true

I am testing out both to confirm, given the "EKS SubnetIds and SecurityIds must coexist or be both empty list" error CloudFormation spits out, is the fix to deploy without ALB first, then enable post deployment? (I was integrating this QS into a Ci pipeline for a development environment, so will need to think about how to handle this, is it a bug?)

**Comment by jaymccon** [comment]
The error message is misleading, I will work on a pre-flight check for subnet tags, so that we can fail early on for subnets without the required tags. 

@kn0xyland afaik the aws cloud controller for kubernetes just looks for the presence of the tag key (`kubernetes.io/role/internal-elb` the value `1`/`true` is arbitrary and can be set to anything really.

From my perspective there is no bug here, before you deploy eks into an existing vpc, you must ensure the subnets are tagged. If this is done there is no need to deploy without alb, and then add alb later.

**Comment by kn0xyland** [comment]
Thanks Jaymccon for your reply!

I have my public and private subnets tagged correctly I think, as I'm running multiple clusters in the same account I have them set to:

VPC1 Subnets: (Running the eks qs v2 stack)
kubernetes.io/cluster/MyCluster1 = owned
kubernetes.io/role/internal-elb = true

VPC2 Subnets: (same account as VPC1, running the eks qs v3 stack)
kubernetes.io/cluster/MyCluster2 = owned
kubernetes.io/role/internal-elb = true

It does mention a SecurityID, could this be throwing it if the subnets are tagged correctly?

**Comment by kn0xyland** [comment]
Update:

I had 3x additional private subnets (6 in total) tagged with internal-elb in my VPC CFN code, when I removed these from my VPC template the ALB controller deployed without incident. 

Thank you for your assistance, and many thanks to the community here :heart:

**Comment by xor007** [comment]
I only get this error if I try to deploy an ENIConfig resource as custom k8s resource using https://github.com/aws-quickstart/quickstart-kubernetes-resource-provider
